tests/test_utils.py

```diff
@@ -208,3 +225,21 @@ def test_obfuscate_sensitive_headers(headers, output):
     bytes_output = [(k.encode(), v.encode()) for k, v in output]
     assert list(obfuscate_sensitive_headers(headers)) == output
     assert list(obfuscate_sensitive_headers(bytes_headers)) == bytes_output
+
+
+@pytest.mark.parametrize(
+    ["url", "no_proxy", "expected"],
+    [
+        ("http://127.0.0.1", {"NO_PROXY": ""}, True),
+        ("http://127.0.0.1", {"NO_PROXY": "127.0.0.1"}, False),
+        ("http://127.0.0.1", {"NO_PROXY": "1.1.1.1"}, True),
+        ("http://courses.mit.edu", {"NO_PROXY": "mit.edu"}, False),
+        ("https://mit.edu.info", {"NO_PROXY": "mit.edu"}, True),
+        ("https://mit.edu.info", {"NO_PROXY": "mit.edu,edu.info"}, False),
+        ("https://mit.edu.info", {"NO_PROXY": "mit.edu,mit.info"}, True),
```

Comment:
Should we include the `NO_PROXY` list case?

Reply:
the list is in the `mit.edu,edu.info` examples